bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -199,4 +202,11 @@ unique_ptr<MessageBase> RequestProcessingState::buildClientRequestMsg(bool empty
   return clientPreProcessReqMsg_->convertToClientRequestMsg(emptyReq);
 }
 
+// Returns const reference to internal state. NOT THREAD SAFE!
```

Comment:
Actually, none of RequestProcessingState functions is thread-safe. Maybe worth adding a general comment that they are guarded by RequestState::mutex from PreProcessor.